low-level/framework/base/global_config.py

```diff
@@ -0,0 +1,104 @@
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+from cortx.utils.conf_store import Conf
+from framework.utils.conf_utils import (GLOBAL_CONF, SSPL_CONF,
+            CLUSTER, SERVER_NODES, MACHINE_ID, STORAGE, STORAGE_SET_ID,
+            ENCLOSURE_ID, CONTROLLER, PRIMARY, IP, PORT, SECONDARY,
+            USER, PRODUCT, RSYSLOG, HOST, NODE_TYPE, TYPE, RELEASE,
+            TARGET_BUILD, SECRET, CONF_API_DELIMITER)
+from framework.base.sspl_constants import (DATACENTER01, RACK01, SERVERNODE01,
+                CLUSTER01, DEFAULT_MC_IP, DEFAULT_SYSLOG_HOST, DEFAULT_SYSLOG_PORT)
+
+
+class GlobalConf:
+    """Base Class to fetch global config values."""
+
+    Conf.load(SSPL_CONF, "yaml:///etc/sspl.conf")
+    global_config = Conf.get(SSPL_CONF, "SYSTEM_INFORMATION>global_config_copy_url")
+    Conf.load(GLOBAL_CONF, global_config)
+    
+    def __init__(self):
+        self.initialize()
+        self.map_global_config_key_values()
+
+    def initialize(self):
+        self.SRVNODE = Conf.get(GLOBAL_CONF, f'{CLUSTER}>{SERVER_NODES}')[MACHINE_ID]
+        self.ENCLOSURE = Conf.get(GLOBAL_CONF, f"{CLUSTER}>{self.SRVNODE}>{STORAGE}>{ENCLOSURE_ID}")
+
+    def fetch_global_config(self, config_list):
+        response = {}
+        default_val = ''
+        req_fields = []
+        for key in config_list:
+            if key not in self.global_conf_dict.keys():
+                return
+            output = self.global_conf_dict[key]
+            for ele in output:
+                if "default:" in ele:
+                    default_val = ele.split(':')[1]
+                else:
+                    req_fields.append(ele)
+            req_string =  CONF_API_DELIMITER.join(req_fields)
+            global_val = Conf.get(GLOBAL_CONF, req_string, default_val)
+            response[key] = global_val
+            req_fields = []
+        return response
+
+    def fetch_sspl_config(self, query_string=False, default_val=""):
+        try:
+            if not query_string:
+                return
+            response = Conf.get(SSPL_CONF, query_string, default_val)
+            return response
+        except Exception as ex:
+            print("Failed to fetch the sspl config:{}".format(str(ex)))
+
+    def set_sspl_config(self, request="", value=False):
+        try:
+            if not value:
+                return
+            response = Conf.set(SSPL_CONF, request, value)
+            return response
+        except Exception as ex:
+            print("Failed to set the sspl config:{}".format(str(ex)))
+
+    def map_global_config_key_values(self):
+        self.global_conf_dict = {
+            "cluster_id": [CLUSTER, 'cluster_id', 'default:' + CLUSTER01],
+            "site_id": [CLUSTER, self.SRVNODE, 'site_id', 'default:' + DATACENTER01],
```

Comment:
Good to see you found a way to keep default values in same table.

so "CLUSTER, self.SRVNODE" holds the key hierarchy and 'site_id' is the actual key right. site_id is the key to this list also, why duplicated in list?
It can stay as 
"site_id": [ CLUSTER + self.SRVNODE, 'default:' + DATACENTER01],

Reply:
currently in fetch_global_config() function,
the requested str is created as:
req_string =  CONF_API_DELIMITER.join(req_fields)
irrespective of the fields.
If we want to keep the dict as:
"site_id": [ CLUSTER + self.SRVNODE, 'default:' + DATACENTER01],
then we need to add additional code to add 'site_id' key in the req_field list before 'default:value' element.
Please suggest, should I do this change?